eng/build.ps1

```diff
@@ -369,82 +369,114 @@ function TestUsingRunTests() {
-  $args += " --dotnet `"$dotnetExe`""
-  $args += " --logs `"$LogDir`""
-  $args += " --configuration $configuration"
+  
+  $testAssemblyFinderArgs += " --artifactsDirectory `"$ArtifactsDir`""
+  $testAssemblyFinderArgs += " --configuration $configuration"
+
+  $testAssembliesPath = Join-Path $ArtifactsDir "testassemblies.txt"
+  $testAssemblyFinderArgs += " --outputFilePath `"$testAssembliesPath`""
 
   if ($testCoreClr) {
-    $args += " --tfm net6.0"
-    $args += " --timeout 90"
+    $testAssemblyFinderArgs += " --targetFrameworks net6.0"
     if ($testCompilerOnly) {
-      $args += GetCompilerTestAssembliesIncludePaths
+      $testAssemblyFinderArgs += GetCompilerTestAssembliesIncludePaths
     } else {
-      $args += " --tfm net6.0-windows"
-      $args += " --include '\.UnitTests'"
+      $testAssemblyFinderArgs += " --targetFrameworks net6.0-windows"
+      $testAssemblyFinderArgs += " --include '\.UnitTests'"
     }
   }
   elseif ($testDesktop -or ($testIOperation -and -not $testCoreClr)) {
-    $args += " --tfm net472"
-    $args += " --timeout 90"
+    $testAssemblyFinderArgs += " --targetFrameworks net472"
 
     if ($testCompilerOnly) {
-      $args += GetCompilerTestAssembliesIncludePaths
+      $testAssemblyFinderArgs += GetCompilerTestAssembliesIncludePaths
     } else {
-      $args += " --include '\.UnitTests'"
+      $testAssemblyFinderArgs += " --include '\.UnitTests'"
     }
 
     if ($testArch -ne "x86") {
-      $args += " --exclude '\.InteractiveHost'"
+      $testAssemblyFinderArgs += " --exclude '\.InteractiveHost'"
     }
 
   } elseif ($testVsi) {
-    $args += " --timeout 110"
-    $args += " --tfm net472"
-    $args += " --retry"
-    $args += " --sequential"
-    $args += " --include '\.IntegrationTests'"
-    $args += " --include 'Microsoft.CodeAnalysis.Workspaces.MSBuild.UnitTests'"
-
-    if ($lspEditor) {
-      $args += " --testfilter Editor=LanguageServerProtocol"
-    }
-  }
-
-  if (-not $ci -and -not $testVsi) {
-    $args += " --html"
+    $testAssemblyFinderArgs += " --targetFrameworks net472"
+    $testAssemblyFinderArgs += " --include '\.IntegrationTests'"
+    $testAssemblyFinderArgs += " --include 'Microsoft.CodeAnalysis.Workspaces.MSBuild.UnitTests'"
   }
 
   if ($collectDumps) {
     $procdumpFilePath = Ensure-ProcDump
-    $args += " --procdumppath $procDumpFilePath"
-    $args += " --collectdumps";
+    EnableRegistryDumpCollection
   }
 
-  $args += " --arch $testArch"
+  # Figure out which assemblies we need to run tests on.
+  Write-Host "$testAssemblyFinder $testAssemblyFinderArgs"
+  Exec-Console $dotnetExe "$testAssemblyFinder $testAssemblyFinderArgs"
 
-  if ($sequential) {
-    $args += " --sequential"
+  $dotnetTestAdditionalArgs = GetCommonDotnetTestArgs
+  
+  $testRunnerExecutableArgs = $null
+  if ($testVsi)
+  {
+    # If we're running integration tests we need to execute tests via RunTests.
+    $runTests = GetProjectOutputBinary "RunTests.dll" -tfm "net6.0"
```

Comment:
Once are entirely on the new integration test framework we remove run tests entirely and run everything (except helix) via dotnet test.